### ISSUE TITLE
Datepicker doesn't react on dateformat change if date was entered by hand

### ISSUE THREAD

**Issue by jmayday** [angular-ui/bootstrap#3291]
Hi.

Datepicker is not presenting a date in new format after dateformat change, when date was entered by hand. 

Let's say we enter date 11.11.2011 having 'dd.MM.yyyy' selected as format. Then after switching format to 'yyyy/MM/dd' nothing hapens (value "11.11.2011" is not updated to "2011/11/11"). If we have some validation on form for dates it also breaks. 

No plunker is required as the problem can be recreated here http://angular-ui.github.io/bootstrap/#/datepicker

**Comment by jmayday** [comment]
You can use this plunker if you want to do experiments with code: http://plnkr.co/edit/yNuSZPiQYFn0Zy2COPdA

**Comment by RobJacobs** [comment]
@jmayday Is this a problem your experiencing in your applications or are you reporting a problem with the documentation implementation?

**Comment by jmayday** [comment]
As I wrote this is possible to reproduce using datepicker example. This poblem is a dateParser service problem. It's not related to any specific application where datepicker is used.

**Comment by jmayday** [comment]
Sorry - I didn't mention that date format change works in case when date was picked from calendar. When date was typed manually date format is not applied after change. I find this beahviour inconsistent.

**Comment by jmayday** [comment]
@RobJacobs sorry but I don't understand your answer. What do you mean by "how to format dates on an input element"? Could you provide some links?

I agree with what you wrote about current behvaior and this is exactly what I meant and in my opinion this behavior is just inconsistent. 

Having plunker I provided or datepicker demo on angular-ui site as example:

- set format of your choice, for example yyyy/MM/dd
- set date from datepicker to 2015/02/16
- change format to some other -> format is successfully applied on entered date
- set format again to yyyy/MM/dd
- set date manually to 2015/02/16
- change format to some other -> format is not applied on entered date

So in both cases date was the same and was correctly formatted before switching format to another one. In my opinion for user this should see no difference in behavior when switching format. No matter if date currently presented was entered by hand or chosen from datepicker (in both cases it's still "2015/02/16"). 

I could adjust this behavior with custom directives, but still - I think it should just work out-of-the-box.

**Comment by metrey** [comment]
It seems that the issue raised here similar to this workaround solution: http://developer.the-hideout.de/?p=119

It's should be added into the core of datepicker here, isn't it?

**Comment by wesleycho** [comment]
I believe this problem should be fixed now, as of a couple of days ago when two major datepicker/dateparser fixes have landed in `master` (& released in 0.13.2 as of last night).
